lightning/src/ln/channel.rs

```diff
@@ -2953,6 +2953,28 @@ impl<Signer: Sign> Channel<Signer> {
 			panic!("Cannot update fee while peer is disconnected/we're awaiting a monitor update (ChannelManager should have caught this)");
 		}
 
+		// Before proposing a feerate update, check that we can actually afford the new fee.
+		let inbound_stats = self.get_inbound_pending_htlc_stats(Some(feerate_per_kw));
+		let outbound_stats = self.get_outbound_pending_htlc_stats(Some(feerate_per_kw));
+		let total_fee = feerate_per_kw as u64 * (COMMITMENT_TX_BASE_WEIGHT + (inbound_stats.pending_htlcs as u64 + outbound_stats.pending_htlcs as u64) * COMMITMENT_TX_WEIGHT_PER_HTLC) / 1000;
+		let holder_balance_after_fee_sub = (self.value_to_self_msat / 1000).checked_sub(total_fee).map(|a| a.checked_sub(outbound_stats.pending_htlcs_value_msat / 1000)).unwrap_or(None).unwrap_or(u64::max_value());
+		if holder_balance_after_fee_sub < self.counterparty_selected_channel_reserve_satoshis.unwrap() {
+			log_debug!(logger, "Cannot afford to send new feerate at {}", feerate_per_kw);
```

Comment:
Might it be worth knowing when we've attempted and failed to update the feerate a bunch of times (i.e. add an auto-close similar to #1056)? And/or, would it improve user safety if we just raise the fee as much as we can afford to here, instead of not updating it at all?

Brainstorming mostly

Reply:
I think that's a really relevant suggestion! Ideally, if we failed too much and we're failing from our feerate efficiency target we should force-close. On the other-side, I know that other implementations are offering a configuration knob to their node operators to waive at that kind of mechanism. Automatic force-close might not make sense if the channel is purely internal between your nodes or opened with a more-or-less trusted counterparty.

Adding a TODO for now and let pursuing the conversation in #1056 ?

Reply:
> Adding a TODO for now and let pursuing the conversation in #1056 ?

Sounds good!

I guess in a way, the *worst* time to force-close is when we're having difficulty raising the fee :thinking: